sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectsShouldNotBeDisposedMoreThanOnce.cs

```diff
@@ -177,22 +179,45 @@ private ProgramState VisitInvocationExpression(InvocationExpressionSyntax instru
             private ProgramState ProcessStreamDisposingTypes(ProgramState programState, SyntaxNode usingExpression)
             {
                 var newProgramState = programState;
+                var objectCreationExpressions = usingExpression.DescendantNodes()
+                    .OfType<ObjectCreationExpressionSyntax>();
 
-                var arguments = usingExpression.DescendantNodes()
-                    .OfType<ObjectCreationExpressionSyntax>()
-                    .Where(IsStreamDisposingType)
-                    .Select(FirstArgumentOrDefault)
-                    .WhereNotNull();
-
-                foreach (var argument in arguments)
+                foreach (var objectCreationExpression in objectCreationExpressions)
                 {
-                    var streamSymbol = this.semanticModel.GetSymbolInfo(argument.Expression).Symbol;
-                    newProgramState = ProcessDisposableSymbol(newProgramState, argument.Expression, streamSymbol);
+                    var objectType = this.semanticModel.GetSymbolInfo(objectCreationExpression.Type)
+                    .Symbol

```

Comment:
also, I think `Symbol` could be null if rolsyn doesn't manage to resolve it (`SymbolInfo` is not null because it's a `struct` but `Symbol` can be null, and we're usually defensive because the analyzer can run on the fly inside the IDE when not everything compiles)